### DISCUSSION TITLE
Preflight does not seem to support direct+manager

### DISCUSSION THREAD

**Discussion by trocade** [Security-Onion-Solutions/securityonion#7018] [Unsupported Versions] [answered]
I just tried to install a 2.3.91 sensor on a host which has no Internet using the "Direct + Manager" option.

The Preflight does not seem to be using this  option and will try to get the repositories directly.

Once preflight failed I continued anyway and installation completes as expected.

Also in the preflight the check it is doing for the remote repos needs an improved timeout.
The current timeout per curl command is 65 seconds (Default 60 seconds for curl and then a 5 second sleep) * 10

The current command is this:
`curl -s -o /dev/null -w %{http_code} -L https://download.docker.com/linux/ubuntu/gpg`

I would suggest:
`curl --connect-timeout 5 -s -o /dev/null -w %{http_code} -L https://download.docker.com/linux/ubuntu/gpg`

The total time for preflight was close to 25 minutes for me until I modified the so-preflight code with the above.

**Comment by rwwiv** [comment, marked as answer]
We don't really support mixed connectivity environments like this. Like you've seen there are ways around it of course, and we're taking a look at further improving setup in future versions based on your feedback.

There's some further reading in our [docs here](https://docs.securityonion.net/en/2.3/airgap.html). I've included the relevant part of the page below as well.

> Please note that the airgap option is intended to be consistent across your deployment, so if you are on an airgap network you should choose the airgap option when installing the manager and all nodes (recent versions enforce this automatically).

**Reply by trocade** [reply]
I should probably be more clear of course, sorry for lack of details.

This is a distributed environment with a manager and sensors, it is not airgapped.
The manager has full internet connectivity and the sensors have no internet connectivity, hence the direct + manager setup.

And it is only the preflight script that has problems, the rest of the setup is working just fine.

**Reply by rwwiv** [reply]
Right, the docs are saying that Security Onion isn't intended to be set up in an environment where some, but not all, machines have internet connectivity. The airgap option, while generally intended for airgap networks, is meant to be used in any case where there is no internet access. 

While what you're doing technically works, it's not actually a configuration that's supported out of the box which is why you're seeing the preflight script fail (and one of the reasons we let you continue if the preflight checks fails).

**Comment by bicfordio** [comment]
Due to similar limitations/requirements as @trocade, I had been doing the above deployment (air gapped forward nodes with internet-access manager) for the last 3 months with no real issues beyond the wait time involved for preflight checks. With that said, it appears that for whatever reason beginning on version 2.3.110, the preflight checks get stuck at 98% and doesn't let me continue with the builds any further.
